react-context/flow.txt

```diff
@@ -0,0 +1,9 @@
+User can sign up: When the user clicks 'sign up' button or anchor tag 'register now', the auth-form.jsx file re-renders via the 'alternateActionHref' function to prepare to accept sign-up information from the user - changing the hash route to #sign-up. 
+
+-When the user has entered their username and password and clicks the "Register" button, the handleSubmit function in authForm.jsx fires and sends a post request to api/auth/sign-up with the entered information. Index.js has been configured to send post requests to auth/sign-up via sql INSERT statements. If there is no current user with the entered username, the new username and password are entered into the database and the password is hashed. 
```

Comment:
`index.js` doesn't send any requests. it does "handle" incoming requests though. it's important to be clear that this is not a request.
```js
app.get('/api/something', (req, res) => {

});
```

it's also worth noting that the hashing of the password happens _before_ the insert.